File: rust/src/vertice.rs
pub struct Vertice {
    pub id: i32,
    pub anterior: i32,
    pub explorado: bool,
    pub adjacencias: Vec<i32>,
    pub inicio: bool,
    pub fim: bool,
}

impl Vertice {
    pub fn novo(indice: i32, anterior: i32) -> Self {
        Vertice {
            id: indice,
            anterior: anterior,
            explorado: false,
            adjacencias: Vec::new(),
            inicio: false,
            fim: false
        }
    }
}

File: python/vertice.py
from __future__ import annotations


class Vertice:
    def __init__(self, id: int, anterior: int) -> None:
        self.id: int = id
        self.anterior: int = anterior
        self.explorado: bool = False
        self.adjacencias: list[int] = []
        self.inicio: bool = False
        self.final: bool = False

    def __str__(self) -> str:
        return str(self.id) + ": " + str(self.adjacencias)
